content/articles/routes-and-navigators-in-flutter-apps/index.md

```diff
@@ -0,0 +1,274 @@
+- Route -A route defines the navigation of the app. It tells which page the navigator takes the user to. Routes made of screens and pages.
+To navigate through apps screen,one can navigate using Navigator.push(),navigator.pop().
+### Types of navigator
+- Using named routes
+One of the best ways to manage many routes is using references. One can refer to a route either by Name, convention, or path.
+To map navigator route one uses builder functions. Material app uses this style to create value for callback methods.
+Illustration using code to show route names
+```dart
+void main() {
+ runApp(MaterialApp(
+    home: MyFirstApp(), //represent the '/' route
+     routes: <String, WidgetBuilder> {
+ '/firstpage': (BuildContext context) => My_page(title: 'firstPage'),
+ '/second': (BuildContext context) => My_Page(title: 'Secondpage'),
+ '/thirdpage': (BuildContext context) => My_Page(title: 'thirdpage'),
+ },
+ ));
+}
+```
+- Using navigator.pop:
+To create a new route to a new page using this method :
+
+```dart
+ Navigator.pop(context);
+```
+- Using navigator.push:
+To create a new route to a new page using this method :
+
+```dart
+ Navigator.push(context);
+```
+### Example to understand this better
+Take a case for an application used by users to stream songs or videos. When a user taps on a song from a list of many songs, the new screen that pops up contains more details about the tapped one.
+
+
+
+### Simple application that uses navigators and routes.
+
+For our application, we will have 3 screens. 
+
+### Step one: Create a new flutter application
+  Open your visual code. On the menu bar click on `view` then `command palette` then click on `new flutter project`. Specify the location where to create your project then specify the name of your project. You can also create using the vs code terminal by typing 
+```cmd
+flutter create name_of_project 
+```
+### Step two: Named routes
+Open main. dart file. Erase its content and replace it with this code. This case uses named routes for all three screens. It contains named routes of all the pages we have in our application.
+```dart
+
+import 'allfiles.dart';
+void main() {
+ runApp(const ThisApp());
+}
+
+class ThisApp extends StatelessWidget {
+ const ThisApp({Key? key}) : super(key: key);
+
+ @override
+ Widget build(BuildContext context) {
+ return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      title: 'First Screen',
+      home: const FirstScreen(),
+      routes: {
+ FirstScreen.routeName: (context) => FirstScreen(),
+ Secondscreen.routeName: (context) => const Secondscreen(),
+ Thirdscreen.routeName: (context) => const Thirdscreen(),
+      },
+    );
+  }
+}
+
+```
+### Step three: Coding firstpage.
```

Comment:
You can give better titles for all the steps. Make them a little bit more descreptive.